framework/src/restart/DataIO.C

```diff
@@ -218,13 +218,11 @@ dataLoad(std::istream & stream, std::string & v, void * /*context*/)
   unsigned int size = 0;
   stream.read((char *) &size, sizeof(size));
 
-  // Read the string
-  char* s = new char[size+1];
-  stream.read(s, sizeof(char)*(size+1));
+  // Resize the string data
+  v.resize(size);
 
-  // Store the string and clean up
-  v = s;
-  delete[] s;
+  // Read the string
+  stream.read(&v[0], sizeof(char) * size);
```

Comment:
I was referring to [this](http://stackoverflow.com/questions/16425338/is-it-safe-to-pass-stdstring-to-c-style-apis) which says that C++03 did not guarantee strings were contiguous in memory (although, how could they *not* be?) but can't vouch for its accuracy or my reading of it.